src/main/java/uk/gov/hmcts/ccd/data/user/JurisdictionsResolverFromIdam.java

```diff
@@ -0,0 +1,28 @@
+package uk.gov.hmcts.ccd.data.user;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import org.springframework.web.context.annotation.RequestScope;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@Qualifier(JurisdictionsResolverFromIdam.QUALIFIER)
+@RequestScope
+public class JurisdictionsResolverFromIdam implements JurisdictionsResolver {
+
+    public static final String QUALIFIER = "default";
+
+    @Override
+    public List<String> getJurisdictionsFromIdam(String[] roles) {
```

Comment:
rather than passing `roles` this implementation should inject 
`CachedUserRepository` and call 
`userRepository.getUserDetails()` internally